src/EditorFeatures/CSharpTest/CompleteStatement/CSharpCompleteStatementCommandHandlerTests.cs

```diff
@@ -0,0 +1,2068 @@
+        public void ArgumentListOfMethodInvocation2()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x$$, y)");
+
+            var expected = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y);$$");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfMethodInvocation3()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x,$$ y)");
+
+            var expected = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y);$$");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfMethodInvocation4()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, $$y)");
+
+            var expected = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y);$$");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfMethodInvocation5()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y$$)");
+
+            var expected = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y);$$");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfMethodInvocation6()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y)$$");
+
+            var expected = code.Replace("$$", ";$$");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfMethodInvocation_MissingParen()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y$$");
+
+            var expected = code.Replace("$$", ";$$");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfMethodInvocation_CommentsAfter()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x$$, y) //Comments");
+
+            var expected = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y);$$ //Comments");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfMethodInvocation_SemicolonAlreadyExists()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x$$, y);");
+
+            var expected = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y);$$;");
```

Comment:
weird as well... i can see arguments both ways.  One arg is we're consistent here in inserting the semicolon.  But it just seems weird that in code that is correct, we're going and inserting a semicolon at a place where we already have one.

Reply:
➡️ This is a minority case included for clarity of behavior and so we are aware if anything changes in the future for this feature. Automatic brace completion inserts closing delimiters, but it typically does not insert a semicolon. The Statement Completion feature helps users type this required character at the correct location with fewer keystrokes than previously required.

Reply:
that seems acceptable to me.  Might be good to put this into the docs for the feature.  Might be one area that's worth calling out as potentially changeable depending on how people feel about things in the future.